django/db/models/functions/text.py

```diff
@@ -2,6 +2,17 @@
 from django.db.models.functions import Coalesce
 
 
+class CharFieldMixin:
+    """
+    Add conversion to char fields. Some databases backends such as MySQL
+    returns bytes instead of strings in some edge cases.
+    """
+    def convert_value(self, value, expression, connection):
+        if self.output_field.get_internal_type() == 'CharField' and isinstance(value, bytes):
```

Comment:
We can also add a database feature flag to run conversion only on MySQL.

Reply:
I've left some [comments](https://code.djangoproject.com/ticket/29251#comment:4) in the ticket. IMO this behavior cannot be changed in the MySQL engine. Personally I'm +1 with this approach, because it doesn't have impact on the overall performance of the MySQL back-end.

Reply:
I'm okay with the solution but "Some databases backends such as MySQL returns bytes instead of strings in some edge cases." feels insufficient. What exactly does "some edge cases" mean -- is there any way to tell if a database function might need this behavior? Are there really possibly other databases that behave the same way?

Reply:
I added more detailed description. I will be grateful for your opinion.